rules-reviewed/camel3/camel2/xml-moved-components.windup.xml

```diff
@@ -239,16 +239,18 @@
         </rule>
         <rule id="xml-moved-components-00014">
             <when>
-                <xmlfile as="dependencies-block" in="pom.xml"
-                         matches="/m:project/m:dependencies[count(m:dependency/m:artifactId[text()='camel-main'])=0]">
-                    <namespace prefix="m" uri="http://maven.apache.org/POM/4.0.0"/>
-                </xmlfile>
-                <javaclass references="org.apache.camel.main.Main">
+                <javaclass references="org.apache.camel.main.Main" as="javaClass">
                     <location>IMPORT</location>
                 </javaclass>
+                <not>
+                    <xmlfile as="dependencies" in="pom.xml"
+                             matches="/m:project/m:dependencies/m:dependency/m:artifactId[text()='camel-main']">
+                        <namespace prefix="m" uri="http://maven.apache.org/POM/4.0.0"/>
+                    </xmlfile>
```

Comment:
Implicit AND with a NOT will produce false negatives.
Please check this issue and comment in order to refactor the rule : https://issues.redhat.com/browse/WINDUPRULE-608?focusedCommentId=14149760&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14149760

Reply:
Hi @jonathanvila , thanks for reviewing!

I've changed it to the way you indicated but the rule now fails to trigger. Any Ideas? 
```
<rule id="xml-moved-components-00014">
            <when>
                <javaclass references="org.apache.camel.main.Main" as="javaClass">
                    <location>IMPORT</location>
                </javaclass>
            </when>
            <perform>
                <iteration over="javaClass">
                    <when>
                        <not>
                            <xmlfile as="dependencies" in="pom.xml"
                                     matches="/m:project/m:dependencies/m:dependency/m:artifactId[text()='camel-main']">
                                <namespace prefix="m" uri="http://maven.apache.org/POM/4.0.0"/>
                            </xmlfile>
                        </not>
                    </when>
                    <perform>
                        <hint title="`org.apache.camel.main.Main` class has been moved" effort="1" category-id="mandatory">
                            <message>The class `org.apache.camel.main.Main` has been moved out of `org.apache.camel:camel-core` and into its own JAR named `org.apache.camel:camel-main`.
                                `org.apache.camel:camel-main` has to be added as a dependency to your project pom.xml file
                            </message>
                            <link href="https://camel.apache.org/manual/latest/camel-3-migration-guide.html#_main_class"
                                  title="Camel 3 - Main class"/>
                        </hint>
                    </perform>
                </iteration>
            </perform>
        </rule>
```

Reply:
@jonathanvila using the `project` tags instead of xmlfile seems to work. Push and rebased. Thanks!